docs/data/sdks/sdk-architecture.md

```diff
@@ -0,0 +1,340 @@
+    - It's easier for develpers to onboard Amplitude and also gain more confidence on data quality
+    - Next-gen Browser, Node.js, React Native SDK use TypeScript. Next-gen Android uses Kotlin. Next-gen iOS uses Swift
+    - Improved the code readability can speed up your the developement cycle
+    - Browser SDK implements tree-shaking technique to reduce the bundle size
+
+- Next-gen SDKs are designed with a solid and **extensible architecture**
+    - Next generation SDKs allow more customization based on your needs. You can implement [a custom logger](./#logger-provider) and [a custom storage](./#storage-provider), [enrich your data](./#enrichment-plugin), and [send your data to other destination](./#destination-plugin) by using Plugins
+
+- Next-gen SDKs has **cleaner and aligned interfaces**
+    - Aligned interfaces benefit using Amplitude SDKs on multiple platforms without memorizing different function names. For example, before next-gen SDKs, to track an event with different properties, you need to use different functions, for example `logEvent()`, `logEventWithTimestamp()`, `logEventWithGroup()`, etc. Right now with next-gen SDKs, you can just use by `track()` with all properties in `EventOptions`
+    
+- Next-gen SDKs are **more reliable and robust**
+    - Next-gen SDKs don't require any external dependencies, which enhances performance, mitigate potential issues caused by dependencies, and makes them more lightweight
+    - Next-gen SDKs use file storage instead of database to improve performance
+    - Next-gen Android SDK removes OkHttp which addresses dependency compatibility issues
+
+## Architecture
+
+The next-gen SDKs share the same architecture and interfaces across platform. This page covers the high-level functionality across all next-gen SDKs. Refer to the individual SDK documentation for more detailed examples.
+
+![Next-gen Architecture](/../assets/images/data-next-gen-sdk-architecture.drawio.svg)
+
+### Client
+
+Amplitude Client stores configurations and is also the entry point for developers to track events. 
+
+Before instrumenting events, you must initialize a SDK client instance using the API key for your Amplitude project. 
+
+```typescript
+// Next-gen Browser SDK example: initialize a Client
+
+import * as amplitude from '@amplitude/analytics-browser';
+
+amplitude.init('API_KEY');
+```
+
+### Configuration
+
+Configuration includes the options to configure Client behavior. 
+
+You can pass a Configuration on Client initialization.
+
+```typescript
+// Next-gen Browser SDK example: initialize a Client with a Configuration
+
+import * as amplitude from '@amplitude/analytics-browser';
+ 
+amplitude.init('API_KEY', 'OPTIONAL_USER_ID', {
+  flushQueueSize: 50, // flush queued events when there are 100 or more
+  flushIntervalMillis: 20000, // flush queued events every 20 seconds
+  useBatch: true //use batch mode with batch API endpoint, `https://api2.amplitude.com/batch`
+});
+```
+
```

Comment:
What about adding a collapsed list of common configuration options here? I think it would be helpful. Since this doc is cross platform we can limit it to options that are common to all, and add a mention to see the individual platform SDK pages for a full list.

Reply:
Added a common config options section based on the spreadsheet https://docs.google.com/spreadsheets/d/1xb63_jcP1xE1Pf_RpcKeakbGL1bib_RVe50MJge3_PU/edit?usp=sharing